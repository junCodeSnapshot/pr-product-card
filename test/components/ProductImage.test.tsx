import React from "react";
import ReactTestRenderer from "react-test-renderer";
import { ProductCart, ProductImage } from "../../src/components";
import { product2 } from "../data/productData";


describe('Tests on ProductImage Component', () => {

    test('Loads the product image propertly', () => {
        const wrapper = ReactTestRenderer.create(
            <ProductCart product={ product2 }>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCart>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('Loads the custom image propertly', () => {
        const wrapper = ReactTestRenderer.create(
            <ProductImage img="../testpath/customimage.png" className="customClass"/> 
        )
        expect(wrapper.toJSON()).toMatchSnapshot();

    });
})