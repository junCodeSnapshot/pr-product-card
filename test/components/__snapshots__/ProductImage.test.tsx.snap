// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on ProductImage Component Loads the custom image propertly 1`] = `
<img
  alt="Coffee Mug"
  className="productImg customClass"
  src="../testpath/customimage.png"
/>
`;

exports[`Tests on ProductImage Component Loads the product image propertly 1`] = `
<div
  className=" productCard undefined"
>
  <img
    alt="Coffee Mug"
    className="productImg undefined"
    src="coffee-mug.png"
  />
</div>
`;
